Mathlib/CategoryTheory/Monoidal/Braided.lean

```diff
@@ -123,6 +129,38 @@ theorem braiding_naturality {X X' Y Y' : C} (f : X ⟶ Y) (g : X' ⟶ Y') :
   rw [tensorHom_def' f g, tensorHom_def g f]
   simp_rw [Category.assoc, braiding_naturality_left, braiding_naturality_right_assoc]
 
+theorem yang_baxter (X Y Z : C) :
+    (α_ X Y Z).inv ≫ ((β_ X Y).hom ▷ Z) ≫ (α_ Y X Z).hom ≫
+    (Y ◁ (β_ X Z).hom) ≫ (α_ Y Z X).inv ≫
+    ((β_ Y Z).hom ▷ X) ≫ (α_ Z Y X).hom
+    = (X ◁ (β_ Y Z).hom) ≫ (α_ X Z Y).inv ≫
+      ((β_ X Z).hom ▷ Y) ≫ (α_ Z X Y).hom ≫
+      (Z ◁ (β_ X Y).hom) := by
+  have := (braiding_naturality_right_assoc X (β_ Y Z).hom (α_ Z Y X).hom).symm
+  refine Eq.trans (Eq.symm ?_) (Eq.trans this ?_)
+  · refine Eq.trans (congrArg (. ≫ _) (braiding_tensor_right X Y Z)) ?_
+    simp only [assoc]
+  · refine congrArg (_ ≫ .) ((Iso.eq_comp_inv _).mp ?_)
+    refine Eq.trans (braiding_tensor_right X Z Y) ?_
+    simp only [assoc]
+
+theorem yang_baxter' (X Y Z : C) :
+    Iso.homCongr (α_ X Y Z) (α_ Z Y X)
+      (((β_ X Y).hom ▷ Z) ⊗≫ (Y ◁ (β_ X Z).hom) ⊗≫ ((β_ Y Z).hom ▷ X))
+    = ((X ◁ (β_ Y Z).hom) ⊗≫ ((β_ X Z).hom ▷ Y) ⊗≫ (Z ◁ (β_ X Y).hom)) := by
+  dsimp only [Mathlib.Tactic.Coherence.monoidalComp,
+              Mathlib.Tactic.Coherence.MonoidalCoherence.hom, Iso.homCongr_apply]
+  simp only [whiskerRight_tensor, id_whiskerRight, id_comp, Iso.inv_hom_id,
+             comp_id, assoc, tensor_id, id_comp, yang_baxter]
```

Comment:
I would think that the following statement would be more in the spirit of the current API:
```suggestion
theorem yang_baxter' (X Y Z : C) :
    ((β_ X Y).hom ▷ Z) ⊗≫ (Y ◁ (β_ X Z).hom) ⊗≫ ((β_ Y Z).hom ▷ X) =
      𝟙 _ ⊗≫ ((X ◁ (β_ Y Z).hom) ⊗≫ ((β_ X Z).hom ▷ Y) ⊗≫ (Z ◁ (β_ X Y).hom)) ⊗≫ 𝟙 _ := by
  rw [← cancel_epi (α_ X Y Z).inv, ← cancel_mono (α_ Z Y X).hom]
  convert yang_baxter X Y Z using 1
  all_goals coherence
```